From pavel@suse.cz  Tue Feb  3 17:05:31 2009
From: Pavel Machek <pavel@suse.cz>
Date: Fri, 30 Jan 2009 10:05:25 +0100
Subject: Staging: w35und: fix registration with wlan stack
To: Sandro Bonazzola <sandro.bonazzola@gmail.com>, Greg KH <greg@kroah.com>
Cc: Pekka Enberg <penberg@cs.helsinki.fi>
Message-ID: <20090130090525.GA1997@elf.ucw.cz>
Content-Disposition: inline


Initialize few more fields in wireless device structure so that
wireless core actually accepts our registration.

Signed-off-by: Pavel Machek <pavel@suse.cz>
Acked-by: Pekka Enberg <penberg@cs.helsinki.fi>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


---
 drivers/staging/winbond/wbusb.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/drivers/staging/winbond/wbusb.c
+++ b/drivers/staging/winbond/wbusb.c
@@ -369,9 +369,11 @@ static int wb35_probe(struct usb_interfa
 	}
 
 	dev->extra_tx_headroom = 12;	/* FIXME */
-	dev->flags = 0;
+	dev->flags = IEEE80211_HW_SIGNAL_UNSPEC;
+	dev->wiphy->interface_modes = BIT(NL80211_IFTYPE_STATION);
 
 	dev->channel_change_time = 1000;
+	dev->max_signal = 100;
 	dev->queues = 1;
 
 	dev->wiphy->bands[IEEE80211_BAND_2GHZ] = &wbsoft_band_2GHz;
